CONTRIBUTING.md

```diff
@@ -0,0 +1,44 @@
+# Contributing
+
+Kubeaddons-Configs is an Addon Repository which manages a predefined set of Addons and their configurations for deployment via [Kubeaddons](https://github.com/mesosphere/kubeaddons).
+
+The [Kubeaddons Contributing Documentation](https://github.com/mesosphere/kubeaddons/blob/master/CONTRIBUTING.md) governs contributions to this repo in addition to other sections of this document.
+
+## PR Readiness
+
+PR's should be marked `wip` until the following are taken care of:
+
+1. Manual testing of the Addon has succeeded against the most recent release of the `kubeaddons` controller
+2. CI must pass
+
+Once these are complete the PR can be marked as `ready`.
+
+## Creating New Addons
+
+You can create new Addons by providing an `Addon` [resource]() in a [manifest]() placed in the `templates/` directory.
+
+See the [Custom Resource Definition](https://github.com/mesosphere/kubeaddons/blob/master/config/crd/bases/kubeaddons.mesosphere.io_addons.yaml) for the `Addon` type for structure, and the [samples](https://github.com/mesosphere/kubeaddons/tree/master/config/samples) for more explicit implementation examples.
+
+The overall workflow for contributing a new Addon is:
+
+1. define an `Addon` resource as discussed above
+2. install the [Kubeaddons Controller]() to your Kubernetes cluster
+3. deploy the new `Addon`
+4. watch the pod logs for the controller, and ensure that the `Addon` status becomes `ready: true`
+5. test the addon functionality as appropriate for its type
+
+1. Manual testing of the Addon has succeeded against the most recent release of the `kubeaddons` controller
+2. CI must pass
+
+## Customizing Existing Addons
+
+Addons can be customized via their manifests in the `templates/` directory.
+
+Changing an addons name or namespace is done in the `metadata` field.
+
+Configuration of the `Addon` resource itself is primarily done via via the `spec` field, particularly changing the [helm values configuration]() of a helm-based addon is done via the `spec.ChartReference.Values` field.
+
+## ExtraSteps
```

Comment:
It is a bit more involved. Does this belong here? I suspect we may want to mention it but the actual details belong in  mesosphere/charts.

Reply:
Mentioning it here is important, but the details of _creating an init container_ should be in `kubeaddons-extrasteps`; perhaps mentioning them in `charts` is also a good idea, since they'll be referenced from there.

Reply:
I'll be adding CONTRIBING.md files to these other two repos as well, I will ensure they cover their angles on this as well :+1: